src/main/java/eventprocessing/storage/MessageLog.java

```diff
@@ -0,0 +1,26 @@
+package eventprocessing.storage;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MessageLog {
+
+    private List<String> messageHistory = new ArrayList<>();
+    private int maxSize;
+
+    public MessageLog(int maxSize) {
+        this.maxSize = maxSize;
+    }
+
+    public List<String> getMessageHistory() {

```

Comment:
Be careful when returning a collection of objects for other parts of your code to modify - it's really easy to lose track of what modifies the list and when.

Having a `truncate` method in this class is a good idea - how could you take the idea further? Is it possible to get rid of the `getMessageHistory` method entirely, so this class is the only thing that modifies the list?